tests/foreman/cli/test_discoveryrule.py

```diff
@@ -110,6 +110,7 @@ def test_positive_create_with_name(self, name, discoveryrule_factory):
         """
         rule = discoveryrule_factory(options={'name': name})
         assert rule.name == name
+        DiscoveryRule.delete({'id': rule.id})
```

Comment:
Isn't this required for other tests of this module like `test_positive_create_with_search` and `test_positive_create_with_hostname` ?

Reply:
No it's not necessary, but we could. The only reason I'm adding the delete here is because this test is parameterized to use Non-ASCII characters in the names.

Reply:
This is cleanup, and should be called using `request.addfinalizer`. If the assert fails, this delete won't execute.

```
rule = discoverrule_factory()
requests.addfinalizer(lambda: DiscoveryRule.delete({'id': rule.id}))
assert rule.name == name
```